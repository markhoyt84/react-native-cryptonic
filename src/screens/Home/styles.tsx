import { StyleSheet } from "react-native";

const styles: any = StyleSheet.create({
    container: {
        backgroundColor: "#FBFAFA",
    },
    loadingIndicator: {
        flex: 1,
        marginTop: 100
    }
});
export default styles;
