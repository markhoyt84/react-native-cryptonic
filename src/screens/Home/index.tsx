import * as React from "react";
import {
    Container,
    Header,
    Title,
    Content,
    Body,
    List
} from "native-base";
import { ActivityIndicator } from 'react-native';

import styles from "./styles";
import AssetRow from "../../components/AssetRow";
export interface Props {
    assets: any;
    viewingCurrentAssetTrades: string;
    recentTradeData: [];
    loadRecentTradeDataForKeyPair: Function;
    loadingAssets: boolean;
}
export interface State {}
class Home extends React.Component<Props, State> {

    _loadRecentTradeData = (forKeyPairName:string) => {
        this.props.loadRecentTradeDataForKeyPair(forKeyPairName);
    };

    render() {
        const assetKeys = Object.keys(this.props.assets);
        return (
            <Container style={styles.container}>
                <Header style={styles.header}>
                    <Body>
                    <Title>Cryptonic Markets</Title>
                    </Body>
                </Header>
                <Content>
                    { this.props.loadingAssets &&
                    <ActivityIndicator
                        size='large'
                        style={styles.loadingIndicator}
                    />
                    }
                    <List>
                        {assetKeys.map((item, i) => (
                            <AssetRow
                                asset={this.props.assets[item]}
                                key={item}
                                loadRecentTradeData={this._loadRecentTradeData}
                                recentTradeData={this.props.recentTradeData}
                                viewingCurrentAssetTrades={this.props.viewingCurrentAssetTrades}
                            />
                        ))}
                    </List>
                </Content>
            </Container>
        );
    }
}

export default Home;
