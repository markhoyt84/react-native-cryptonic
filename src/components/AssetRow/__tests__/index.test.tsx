import * as React from "react";
import "react-native";
import Home, {default as AssetRow} from "../index";

import * as renderer from "react-test-renderer";

const loadRecentTradeData = jest.fn();
const viewingCurrentAssetTrades = "XXRPXXBT";
const key = "XXRPXXBT";
const asset = {
    XXRPXXBT: {
            altname: 'XRPXBT',
            wsname: 'XRP/XBT',
            aclass_base: 'currency',
            base: 'XXRP',
            marketData:
                {
                    a: [ '0.000064540', '82', '82.000' ],
                    b: [ '0.000064520', '133', '133.000' ],
                    c: [ '0.000064520', '78.24731849' ],
                    v: [ '62230.34154864', '2146763.84366635' ],
                    p: [ '0.000064451', '0.000063446' ],
                    t: [ 70, 1343 ],
                    l: [ '0.000064190', '0.000061670' ],
                    h: [ '0.000064890', '0.000065460' ],
                    o: '0.000064220'
                }
    },
    XXRPZCAD: {
            altname: 'XRPCAD',
            wsname: 'XRP/CAD',
            aclass_base: 'currency',
            base: 'XXRP',
            marketData:
                {
                    a: [ '0.45965000', '295', '295.000' ],
                    b: [ '0.45650000', '133', '133.000' ],
                    c: [ '0.46445000', '560.48311500' ],
                    v: [ '3614.00475523', '148109.70996547' ],
                    p: [ '0.45780850', '0.44328552' ],
                    t: [ 10, 211 ],
                    l: [ '0.45364000', '0.43540000' ],
                    h: [ '0.46445000', '0.46445000' ],
                    o: '0.45461000'
                }
    }
};
const recentTradeData = [];

it("renders correctly", () => {
    const tree = renderer.create(<AssetRow asset={asset} key={key} loadRecentTradeData={loadRecentTradeData} recentTradeData={[]} viewingCurrentAssetTrades={viewingCurrentAssetTrades}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
