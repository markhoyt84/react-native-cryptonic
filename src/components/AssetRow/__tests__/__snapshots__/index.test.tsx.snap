// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FBFAFA",
      "padding": 10,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    style={
      Object {
        "fontSize": 20,
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    Asset Name: 
    XXRP
  </Text>
  <Text
    style={
      Object {
        "paddingBottom": 5,
        "paddingTop": 5,
      }
    }
  >
    CurrentTradingPair: 
    XXRPXXBT
  </Text>
  <Text
    style={
      Object {
        "paddingBottom": 5,
        "paddingTop": 5,
      }
    }
  >
    Opening Price: $
    0.000064220
  </Text>
  <View
    removeClippedSubviews={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      accessibilityLabel="XXRPXXBT"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0076FF",
          "borderBottomLeftRadius": 5,
          "borderColor": "#0076FF",
          "borderRightWidth": 1,
          "borderTopLeftRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
          "paddingVertical": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#0076FF",
              },
              Object {},
              Array [
                Object {
                  "color": "white",
                },
                Object {},
              ],
            ]
          }
        >
          XXRPXXBT
        </Text>
      </View>
    </View>
    <View
      accessibilityLabel="XXRPZCAD"
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomRightRadius": 5,
          "borderColor": "#0076FF",
          "borderLeftWidth": 0,
          "borderTopRightRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
          "paddingVertical": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#0076FF",
              },
              Object {},
              Object {},
            ]
          }
        >
          XXRPZCAD
        </Text>
      </View>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        View Recent Trades
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
